src/app/scam/components/shared/template-launch/template-launch.component.ts

```diff
@@ -14,39 +14,60 @@ import { FormSshKeyComponentModule } from '../form/form-ssh-key/form-ssh-key.com
 import { MatFormFieldModule } from '@angular/material/form-field';
 import { MatInputModule } from '@angular/material/input';
 import { MatButtonModule } from '@angular/material/button';
+import { SurveyService } from '../../../../core/services/survey.service';
+import { Survey } from '../../../../core/interfaces/survey';
+import { SurveySendComponentModule } from '../../templates/survey-send/survey-send.component';
+import { SurveyInputManagerService } from '../../../../core/services/survey-input-manager.service';
 
 @Component({
   selector: 'app-template-launch',
   templateUrl: './template-launch.component.html',
 })
-export class TemplateLaunchComponent implements OnChanges {
+export class TemplateLaunchComponent implements OnInit, OnChanges {
   @Input() template: Template;
   @Output() cancelEvent = new EventEmitter<void>();
 
-  form = new TemplateForm();
+  public form = new TemplateForm();
+  public survey: Survey;
 
-  constructor(private jobsService: JobsService, private router: Router) {
+  constructor(
+    private jobsService: JobsService,
+    private router: Router,
+    private surveyService: SurveyService,
+    private surveyInputManager: SurveyInputManagerService
+  ) {
+  }
+
+  ngOnInit(): void {
+    this.surveyInputManager.inputSource.subscribe((extraVariables: string) => {
+      this.form.ExtraVariables.setValue(extraVariables);
```

Comment:
I think extra variables from the survey shouldn't overwrite original extra variables. It should be merged.